### PR TITLE
Phase 6: Asinh Velocity Transform — Compression on Ux/Uy Channels

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current asinh transform applies only to the pressure channel (index 2 of the 3-channel output). Velocity channels (Ux at index 0, Uy at index 1) are left untransformed. However, velocity in CFD wake flows also has highly non-Gaussian distributions: near-surface boundary layers have steep gradients, the far-field has near-zero velocity. Uy (cross-stream) has strong antisymmetry around the chord line.

Applying asinh compression to velocity channels reduces the effective dynamic range and makes the L1 loss less dominated by high-velocity regions — the same argument that motivated pressure asinh (PR #2054, p_oodc -6.1%).

### Code changes (~10 lines)

1. Find where the pressure asinh transform is applied in train.py. It should look something like:
\`\`\`python
_yp[:, :, 2:3] = torch.asinh(_yp[:, :, 2:3] * cfg.asinh_scale)
\`\`\`

2. Add a new flag and apply asinh to velocity channels:
\`\`\`python
# New config flags:
asinh_velocity: bool = False
asinh_velocity_scale: float = 0.5  # separate scale for velocity (different range than pressure)
\`\`\`

3. When `asinh_velocity=True`, also transform velocity channels:
\`\`\`python
if cfg.asinh_velocity:
    vel_scale = cfg.asinh_velocity_scale
    _yp[:, :, 0:2] = torch.asinh(_yp[:, :, 0:2] * vel_scale)
\`\`\`

4. Apply the INVERSE transform when computing predictions (postprocessing):
\`\`\`python
if cfg.asinh_velocity:
    vel_scale = cfg.asinh_velocity_scale
    pred[:, :, 0:2] = torch.sinh(pred[:, :, 0:2]) / vel_scale
\`\`\`

5. Verify the inverse is correct: `sinh(asinh(x * s)) / s = x` ✓

### Experiment matrix (8 GPUs)

| GPU | Config | Seed | Notes |
|-----|--------|------|-------|
| 0 | Baseline (no velocity asinh) | 42 | Control |
| 1 | Baseline (no velocity asinh) | 43 | Control |
| 2 | `--asinh_velocity --asinh_velocity_scale 0.3` | 42 | Mild compression |
| 3 | `--asinh_velocity --asinh_velocity_scale 0.3` | 43 | Mild compression |
| 4 | `--asinh_velocity --asinh_velocity_scale 0.5` | 42 | Medium compression |
| 5 | `--asinh_velocity --asinh_velocity_scale 0.5` | 43 | Medium compression |
| 6 | `--asinh_velocity --asinh_velocity_scale 1.0` | 42 | Strong compression |
| 7 | `--asinh_velocity --asinh_velocity_scale 1.0` | 43 | Strong compression |

Use `--wandb_group phase6/asinh-velocity` for all runs.

**Base command:**
\`\`\`bash
python train.py --agent alphonse --wandb_group phase6/asinh-velocity \
  --wandb_name "alphonse/asinh-vel-s${scale}-s${seed}" \
  --asinh_velocity --asinh_velocity_scale ${scale} \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --seed ${seed}
\`\`\`

### Important notes
- DO NOT modify MAX_EPOCHS (500) or MAX_TIMEOUT (180.0) in train.py
- The velocity scale should be SEPARATE from the pressure scale (0.75) — different physical ranges
- Start with scale 0.5 (smaller than pressure 0.75) since velocity range may differ
- Watch for divergence — if asinh compresses too much, gradients may vanish
- Also report velocity MAE metrics if available — we care about surface pressure MAE but velocity quality matters for physical consistency

## Baseline

Current best (8-seed ensemble, PR #2080):
- p_in: **12.2** | p_oodc: **6.7** | p_tan: **29.1** | p_re: **5.8**

Single-model baseline (8-seed mean):
- p_in: **13.03** | p_oodc: **7.83** | p_tan: **30.29** | p_re: **6.45**

## Reporting

Post results with individual metrics for all 8 runs, mean ± std per scale, and W&B run IDs.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT alphonse:

## Results

All 8 runs completed at the 180-min wall-clock limit. All runs were still improving at timeout — every final epoch produced a new best checkpoint.

### Surface MAE — Pressure (mae_surf_p, lower=better)

| Run | W&B ID | Seed | Epoch | p_in | p_tan | p_oodc | p_re | val/loss |
|-----|--------|------|-------|------|-------|--------|------|----------|
| baseline | p7iy87v0 | 42 | 157 | 13.5 | 30.4 | 8.1 | 6.4 | 0.3864 |
| baseline | hwzd5zqm | 43 | 157 | 12.9 | 29.8 | 7.8 | 6.4 | 0.3844 |
| **baseline avg** | | | | **13.2** | **30.1** | **7.95** | **6.40** | **0.3854** |
| asinh-0.3 | 2joautz7 | 42 | 156 | 13.6 | 31.2 | 9.0 | 6.8 | 0.4141 |
| asinh-0.3 | g5r2dbxt | 43 | 157 | 13.4 | 31.3 | 8.4 | 6.6 | 0.4047 |
| **asinh-0.3 avg** | | | | **13.5** | **31.25** | **8.70** | **6.70** | **0.4094** |
| asinh-0.5 | 4my1b7qi | 42 | 156 | 13.3 | 32.5 | 8.0 | 6.5 | 0.4002 |
| asinh-0.5 | vye8pe7r | 43 | 157 | 13.9 | 31.0 | 7.9 | 6.6 | 0.4003 |
| **asinh-0.5 avg** | | | | **13.6** | **31.75** | **7.95** | **6.55** | **0.4003** |
| asinh-1.0 | 793xhqav | 42 | 157 | 12.8 | 32.4 | 7.6 | 6.4 | 0.3923 |
| asinh-1.0 | x22wlmnl | 43 | 157 | 13.5 | 33.9 | 7.9 | 6.3 | 0.4045 |
| **asinh-1.0 avg** | | | | **13.15** | **33.15** | **7.75** | **6.35** | **0.3984** |

### vs. Baseline (delta, negative=better)

| Variant | p_in | p_tan | p_oodc | p_re | val/loss |
|---------|------|-------|--------|------|----------|
| asinh-0.3 | +0.3 | +1.15 | +0.75 | +0.30 | +0.0240 |
| asinh-0.5 | +0.4 | +1.65 | 0.00 | +0.15 | +0.0149 |
| asinh-1.0 | -0.05 | +3.05 | -0.20 | -0.05 | +0.0130 |

### vs. 8-Seed Baseline Mean (from BASELINE.md: p_in=13.03, p_tan=30.29, p_oodc=7.83, p_re=6.45)

Our 2-seed baselines track the established baseline well (p_in 13.2 vs 13.03, p_tan 30.1 vs 30.29, p_oodc 7.95 vs 7.83, p_re 6.40 vs 6.45), confirming valid experimental setup.

### Commands Used

```bash
# All runs include these base flags:
BASE="--agent alphonse --wandb_group phase6/asinh-velocity \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3"

# GPU 0-1: baseline (seeds 42, 43)
# GPU 2-3: --asinh_velocity --asinh_velocity_scale 0.3 (seeds 42, 43)
# GPU 4-5: --asinh_velocity --asinh_velocity_scale 0.5 (seeds 42, 43)
# GPU 6-7: --asinh_velocity --asinh_velocity_scale 1.0 (seeds 42, 43)
```

### Peak Memory
38.0 GB (consistent across all 8 runs)

### What Happened

**The asinh velocity transform does not improve surface pressure MAE at any scale tested.** All variants degraded `val_tandem_transfer` (the most important generalization split), and the degradation scaled *inversely* with the compression scale:

- **scale=0.3** (strongest compression): worst overall, +0.75 p_oodc, +1.15 p_tan
- **scale=0.5**: moderate degradation, matched baseline on p_oodc but +1.65 p_tan
- **scale=1.0** (weakest compression): closest to baseline on p_in/p_oodc/p_re (within noise), but worst p_tan at +3.05

The consistent hurt on `val_tandem_transfer` is the clearest signal. This suggests that the asinh transform distorts the velocity feature distribution in a way that specifically hurts tandem geometry generalization. The tandem configuration has complex wake interactions where velocity gradients carry critical information — compressing these gradients via asinh likely removes discriminative signal the model needs.

The baseline's linear normalization appears well-suited for the velocity range in this dataset. Unlike pressure (which benefited from asinh due to extreme outliers), velocity channels don't have the same outlier problem.

**Conclusion: Negative result. The asinh velocity transform hypothesis does not hold. Do not merge.**

### Suggested Follow-ups

1. **Investigate whether velocity dynamic range is actually a problem.** Plot Ux/Uy histograms before and after physics normalization. The pressure asinh worked because pressure had extreme outliers — velocity may not have the same distribution issue.
2. **Learnable per-channel normalization.** Instead of fixed asinh, let the model learn an optimal nonlinear transform (e.g., parametric activation applied to input channels).
3. **Selective velocity transform.** Only transform Uy (cross-stream, more antisymmetric) and leave Ux (streamwise) linear.
4. **Soft clipping / Winsorization.** If specific velocity outliers are the problem, clip rather than compress the entire distribution.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2098 because ADVISOR: **Closing — negative result confirmed.**

Clean experiment, alphonse. The results clearly show that asinh compression on velocity channels hurts performance, especially p_tan (+1.15 to +3.05 across all scales). Your analysis is spot-on:

1. **Velocity doesn't have the outlier problem that pressure has.** The pressure asinh worked because pressure has extreme peaks in stagnation regions. Velocity distributions are more well-behaved after physics normalization.
2. **The p_tan degradation is the clearest signal** — tandem wake interactions rely on velocity gradient information that asinh compression destroys.
3. **The inverse scale relationship** (stronger compression = worse on p_oodc but better on p_tan, weaker compression = opposite) suggests velocity channels carry information at different scales for different flow regimes.

**Key takeaway:** Leave velocity channels in their current linear normalization. The data representation is already well-suited for the velocity range.

Moving you to a fresh hypothesis..
